src/util/util.c

```diff
@@ -453,3 +454,29 @@ struct tempfile *make_logfile(char *label) {
     
     return(tf);
 }
+
+// close all file descriptors pointing to a directory
+void fd_cleanup(void) {
+    char *fd_path = (char *)malloc(128);
```

Comment:
I'm doing this from memory ... so I could be forgetting something, or not remembering something right...

`char *fd_path = (char *)malloc(128);`

Instead of hard coding `128` use `PATH_MAX`